File: solve/lua/select.lua
require "common" 

skeleton(function ()
    for i = 1, n - 1 do
        local min = i
        for j = i + 1, n do
            if a[j] < a[min] then
                local min = j
            end 
        end     
        swap(i, min)
    end     
end)

File: solve/cpp/select.cc
#include "common.hh"

void selectsort(int *beg, int *end)
{
    while (beg != end - 1)
    { 
        int *min = beg;
        for (int *p = beg + 1; p != end; ++p) 
            if (*p < *min) min = p;
        swapp(beg++, min);
    } 
}

int main(int argc, char *argv[])
{
    if (init(argc, argv) == 0)
    {
        selectsort(a, a + n);
        output();
        return 0;
    }
    return 1;
}
